C++/C++.sublime-syntax

```diff
@@ -582,6 +582,16 @@ contexts:
                 - include: global
             - include: expressions
 
+  test-invalid-module-partition:
+    - match: \s*(\:)
+      captures:
+        1: invalid.illegal.unexpected-character.c++
+      set:
+        - match: '.*$'
+          pop: true
+    - match: ''
+      pop: true
```

Comment:
Alternatively the following pattern would scope any number of tokens looking like partitions, but pop the whole module context if something else appears. It's sometimes a good idea to fail as early as possible.

```suggestion
  test-invalid-module-partition:
    - match: (\:)[^:;\s]*
      captures:
        1: invalid.illegal.unexpected-character.c++
    - match: (?=\S)
      pop: true
```

Noone would probably write the following, but it is valid code and should be highlighted properly.

![grafik](https://user-images.githubusercontent.com/16542113/149191193-694bfe36-181e-469c-b734-6e7f4af7528b.png)

Even if the semicolon is missing, keywords should cause the context to pop as well, which is not yet the case.

![grafik](https://user-images.githubusercontent.com/16542113/149191438-ae3cc47a-80e7-4c09-b093-bb8a02717f5a.png)

Reply:
Got it.  I'll handle popping the whole context even if the semi-colon is missing.